modules/widgets/social-icons.php

```diff
@@ -0,0 +1,500 @@
+		$title = apply_filters( 'widget_title', empty( $instance['title'] ) ? '' : $instance['title'], $instance, $this->id_base );
+
+		echo $args['before_widget'];
+
+		if ( ! empty( $title ) ) {
+			echo $args['before_title'] . esc_html( $title ) . $args['after_title'];
+		}
+
+		if ( ! empty( $instance['icons'] ) ) :
+
+			// Get supported social icons.
+			$social_labels = jetpack_social_menu_social_links_labels();
+			$social_icons  = jetpack_social_menu_social_links_icons();
+		?>
+
+			<ul class="jetpack-social-widget-list size-<?php echo esc_attr( $instance['icon-size'] ); ?>">
+
+				<?php foreach ( $instance['icons'] as $icon ) : ?>
+
+					<?php if ( ! empty( $icon['url'] ) ) : ?>
+						<li class="jetpack-social-widget-item">
+							<a href="<?php echo esc_url( $icon['url'], $this->protocols ); ?>">
+								<?php
+									// Add a label if there is supported URL.
+									foreach ( $social_labels as $attr => $value ) {
+										if ( false !== strpos( $icon['url'], $attr ) ) {
+											echo '<span class="screen-reader-text">' . esc_attr( $value ) . '</span>';
+										}
+									}
+
+									$icon_output = jetpack_social_menu_get_svg( array( 'icon' => 'chain' ) );
+
+									// Change SVG icon inside social links menu if there is supported URL.
+									foreach ( $social_icons as $attr => $value ) {
+										if ( false !== strpos( $icon['url'], $attr ) ) {
+											$icon_output = jetpack_social_menu_get_svg( array( 'icon' => esc_attr( $value ) ) );
+										}
+									}
+
+									echo $icon_output;
+								?>
+							</a>
+						</li>
+					<?php endif; ?>
+
+				<?php endforeach; ?>
+
+			</ul>
+
+		<?php
+		endif;
+
+		echo $args['after_widget'];
+	}
+
+	/**
+	 * Sanitize widget form values as they are saved.
+	 *
+	 * @see WP_Widget::update()
+	 *
+	 * @param array $new_instance Values just sent to be saved.
+	 * @param array $old_instance Previously saved values from database.
+	 *
+	 * @return array Updated safe values to be saved.
+	 */
+	public function update( $new_instance, $old_instance ) {
+		$instance['title']     = sanitize_text_field( $new_instance['title'] );
+		$instance['icon-size'] = $this->defaults['icon-size'];
+
+		if ( in_array( $new_instance['icon-size'], array( 'small', 'medium', 'large' ) ) ) {
+			$instance['icon-size'] = $new_instance['icon-size'];
+		}
+
+		$icon_count        = count( $new_instance['url-icons'] );
+		$instance['icons'] = array();
+
+		for ( $i = 0; $i < $icon_count; $i++ ) {
+			$url = filter_var( $new_instance['url-icons'][ $i ], FILTER_SANITIZE_URL );
+
+			if ( ! empty( $url ) ) {
+				$instance['icons'][] = array(
+					'url' => $url,
+				);
+			}
+		}
+
+		return $instance;
+	}
+
+	/**
+	 * Back-end widget form.
+	 *
+	 * @see WP_Widget::form()
+	 *
+	 * @param array $instance Previously saved values from database.
+	 *
+	 * @return string|void
+	 */
+	public function form( $instance ) {
+		$instance = wp_parse_args( (array) $instance, $this->defaults );
```

Comment:
Again, why cast to an array?  wp_parse_args is designed to accept `arg=val&arg2=val2` and the like.  This would break that.